CMakeLists.txt

```diff
@@ -230,26 +209,43 @@ if(NOT MSVC)
 	check_and_add_flag(INIT_SELF -Winit-self)
 	check_and_add_flag(MISSING_DECLARATIONS -Wmissing-declarations)
 	check_and_add_flag(MISSING_VARIABLE_DECLARATIONS -Wmissing-variable-declarations)
-endif(NOT MSVC)
 
-# gcc uses some optimizations which might break stuff without this flag
-add_definitions(-fno-strict-aliasing -fno-exceptions)
+	# gcc uses some optimizations which might break stuff without this flag
+	check_and_add_flag(NO_STRICT_ALIASING -fno-strict-aliasing)
+	check_and_add_flag(NO_EXCEPTIONS -fno-exceptions)
 
-check_and_add_flag(VISIBILITY_INLINES_HIDDEN -fvisibility-inlines-hidden)
-
-if(UNIX AND NOT APPLE)
+	check_and_add_flag(VISIBILITY_INLINES_HIDDEN -fvisibility-inlines-hidden)
 	check_and_add_flag(VISIBILITY_HIDDEN -fvisibility=hidden)
-endif()
 
-if(ENABLE_LTO)
-	check_and_add_flag(LTO -flto)
-	if(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
-		set(CMAKE_AR gcc-ar)
-		set(CMAKE_RANLIB gcc-ranlib)
+	set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fno-pie")
+	if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+		include(CheckCCompilerFlag)
+		check_c_compiler_flag("-no-pie" NO_PIE_UPSTREAM)
+		if(NO_PIE_UPSTREAM)
+			set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -no-pie")
+		endif()
+		check_c_compiler_flag("-nopie" NO_PIE_PATCHED)
+		if(NO_PIE_PATCHED)
+			set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -nopie")
+		endif()
 	endif()
 endif()
 
-if(APPLE)
+if(CMAKE_SYSTEM_NAME MATCHES "Windows")
+	# Only MSBuild needs this, other generators will compile one file at a time
+	if(CMAKE_GENERATOR MATCHES "Visual Studio")
```

Comment:
But /MP is a flag specific to cl.exe only, and the only generator that makes use of it is MSBuild.
Other generators have a separate compiler spawn for each file.